core/kazoo_documents/src/kzd_box_message.erl

```diff
@@ -120,23 +123,55 @@ new(AccountId, Props) ->
                               ),
     Description = props:get_value(<<"Description">>, Props, <<"voicemail message with media">>),
 
-    DocProps = [{<<"_id">>, MsgId}
+    DocProps = props:filter_undefined(
```

Comment:
The `filter_undefined` is done in `kz_json:from_list/1` so no need to add it here and recurse the list twice.